src/propagation.cpp

```diff
@@ -35,23 +36,49 @@ bool has_prefix(const std::string &str, const std::string &prefix) {
 
 }  // namespace
 
+OptionalSamplingPriority asSamplingPriority(int i) {
+  if (i < -1 || i > 2) {
```

Comment:
Not that the values change very often, but it'd be nice if these weren't magic numbers, and the expression was (for example)
`if (i < MinDefinedSamplingPriority || i > MaxDefinedSamplingPriority)`

Reply:
Found a way to define these in the enum itself, so that if it does ever get changed it's more likely the constants will get changed too.